src/main/java/org/embulk/input/FtpFileInputPlugin.java

```diff
@@ -473,7 +469,11 @@ public void run()
         return Channels.newInputStream(t.getReaderChannel());
     }
 
-    private static class FtpInputStreamReopener
+    static int RETRY_LIMIT = 3;
+    static int INITIAL_RETRY_WAIT = 500;
+    static int MAX_RETRY_WAIT = 30*1000;
```

Comment:
I think it is not good idea to make those variables variable... For now, why don't you to add `@VisibleForTesting` annotation to all constants and add a TODO comment saying that those variables are modifyable only for testing and should be configurable at `transaction` method in the future.